rust/src/answer/json.rs

```diff
@@ -57,10 +57,73 @@ impl fmt::Display for JSON {
                 }
                 f.write_char(']')?;
             }
-            JSON::String(string) => write!(f, "{:?}", string.as_ref())?,
+            JSON::String(string) => write_escaped_string(string, f)?,
             JSON::Number(number) => write!(f, "{number}")?,
             JSON::Boolean(boolean) => write!(f, "{boolean}")?,
         }
         Ok(())
     }
 }
+
+fn write_escaped_string(string: &str, f: &mut fmt::Formatter<'_>) -> fmt::Result {
```

Comment:
Let's put a comment here where we took inspiration from so that we can go back there in the future easily if something breaks/changes? Also for creditation (check their license is complete open too please!)

Reply:
Hmm this code has very little in common with serde-json (Apache/MIT, not that this algorithm is their creation anyway). I would've thought testing against their implementation would be enough to give a place to start looking if smth breaks?

Reply:
Ok if it's not copied out from them let's leave it as is